sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -695,7 +695,7 @@ class Analyzer(
           (oldVersion, oldVersion.copy(aggregateExpressions = newAliases(aggregateExpressions)))
 
         case oldVersion: Generate
-            if oldVersion.generatedSet.intersect(conflictingAttributes).nonEmpty =>
+            if oldVersion.producedAttributes.intersect(conflictingAttributes).nonEmpty =>
```

Comment:
Is this correct? `AttributeSet.intersect` is special.

Reply:
well, the implementation was identical in class Generate:
  def generatedSet: AttributeSet = AttributeSet(generatorOutput)
  override def producedAttributes: AttributeSet = AttributeSet(generatorOutput)

Reply:
ah i see